client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/multimap/operations/MultimapOperationsFactory.java

```diff
@@ -0,0 +1,102 @@
+package org.infinispan.client.hotrod.impl.multimap.operations;
+
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.infinispan.client.hotrod.Flag;
+import org.infinispan.client.hotrod.RemoteCacheManager;
+import org.infinispan.client.hotrod.configuration.Configuration;
+import org.infinispan.client.hotrod.impl.operations.HotRodOperation;
+import org.infinispan.client.hotrod.impl.protocol.Codec;
+import org.infinispan.client.hotrod.impl.transport.TransportFactory;
+
+import net.jcip.annotations.Immutable;
+
+/**
+ * Factory for {@link HotRodOperation} objects on Multimap.
+ *
+ * @author karesti@redhat.com
+ * @since 9.2
+ */
+@Immutable
+public class MultimapOperationsFactory {
+
+   private final ThreadLocal<Integer> flagsMap = new ThreadLocal<>();
```

Comment:
I personally am not a big fan of the way these flags worked in the original Client. I would say instead to store the flags on each instance of the RemoteMultiMap and pass them when they invoke something. When a new flag is added or removed we would create a new instance of the MultiMap. This seems much more intuitive to me and simpler.

Then the flags can be defaulted on the RemoteMultiMap when it is retrieved from the manager as well.

Reply:
Do you mind if I leave it like that, as the operations factory, and we try to do a global refactoring in another ticket for both ?

Reply:
Well I was thinking this way would do it properly and we could refactor the other later. But make sure a JIRA is created if you wanted to do this one later.

Reply:
ok